ipc.c

```diff
@@ -0,0 +1,64 @@
+#define _POSIX_C_SOURCE 200809L
+#include <limits.h>
+#include <varlink.h>
+
+#include "kanshi.h"
+#include "ipc.h"
+
+static  void reload_config_done(void *data, struct wl_callback *callback,
+		uint32_t serial) {
+	VarlinkCall *call = data;
+	varlink_call_reply(call, NULL, 0);
+	wl_callback_destroy(callback);
+}
+
+static struct wl_callback_listener reload_config_listener = {
+	.done = reload_config_done
+};
+
+static long handle_reload(VarlinkService *service, VarlinkCall *call,
+		VarlinkObject *parameters, uint64_t flags, void *userdata) {
+	struct kanshi_state *state = userdata;
+	kanshi_reload_config(state);
+	// ensure that changes are applied on the server before sending the response
+	struct wl_callback *callback = wl_display_sync(state->display);
```

Comment:
Ah, I see. This only ensures that the server has received the configuration request, the server is free to wait an arbitrary amount of time before applying the configuration.

We can add a TODO about using the wlr-output-management event instead.